botocore/args.py

```diff
@@ -771,3 +782,40 @@ def _compute_user_agent_appid_config(self, config_kwargs):
                 f'maximum length of {USERAGENT_APPID_MAXLEN} characters.'
             )
         config_kwargs['user_agent_appid'] = user_agent_appid
+
+    def _compute_checksum_config(self, config_kwargs):
+        self._handle_checksum_config(
+            config_kwargs,
+            config_key="request_checksum_calculation",
+            default_value="when_supported",
+            valid_options_enum=RequestChecksumCalculation,
+        )
+        self._handle_checksum_config(
+            config_kwargs,
+            config_key="response_checksum_validation",
+            default_value="when_supported",
+            valid_options_enum=ResponseChecksumValidation,
+        )
+
+    def _handle_checksum_config(
+        self,
+        config_kwargs,
+        config_key,
+        default_value,
+        valid_options_enum,
+    ):
+        value = config_kwargs.get(config_key)
+        if value is None:
+            value = (
+                self._config_store.get_config_variable(config_key)
+                or default_value
+            )
+        value = value.lower()
+        valid_options = valid_options_enum.values()
```

Comment:
I know we talked about this a little previously. I'm curious what value we're getting out of using these enums vs a validation list like we do elsewhere in the codebase. Given we expect users to be providing strings for configuration, is the extra scaffolding we're adding here to create enums a net-benefit or is this additional overhead without a clear use case?

Reply:
I liked the idea of using an Enum since this it would consolidate the valid options for each config. It would prevent some issues like typos when using these options on both the user and maintainer side. However, looking at this now, I don't think this provides enough value and introduces a new way to specify config options which can be more confusing to users. I can update the PR to move away from using the Enums.